stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1803,6 +1800,29 @@ impl NakamotoChainState {
         ))
     }
 
+    /// Get the aggregate public key for a block
+    pub fn get_aggregate_public_key<SH: SortitionHandle>(
+        chainstate: &mut StacksChainState,
+        sortdb: &SortitionDB,
+        sort_handle: &SH,
+        block: &NakamotoBlock,
+    ) -> Result<Point, ChainstateError> {
+        let block_sn =
+            SortitionDB::get_block_snapshot_consensus(sortdb.conn(), &block.header.consensus_hash)?
+                .ok_or(ChainstateError::DBError(DBError::NotFoundError))?;
+        let aggregate_key_block_header =
+            Self::get_canonical_block_header(chainstate.db(), sortdb)?.unwrap();
+
```

Comment:
Can you add a todo comment here with a reference to #4109 ? As currently implemented here (which I think is fine for now) the `at_block` is variable: when a new stacks block is received, `get_canonical_block_header()` returns a different Stacks block, meaning this `get_aggregate_public_key()` will return different values for the same `block` at different times.

Reply:
I'm actually no longer sure that #4109 is desirable, if we're going to allow stackers to re-try DKG in the middle of the cycle.  If that were the case, then we'd want the current behavior.

Reply:
Maybe so, but that merits discussion and that discussion shouldn't stall this PR. Leaving a pointer to that issue makes it so that it's easy to find the code that assumes one approach or the other.

As written, this function may be somewhat dangerous -- I don't think it's strange for a developer to believe that the expected aggregate key for a given block is always the same, but this function will return different results at different times.